app/ai/voice/tts/sarvam.py

```
@@ -44,21 +105,61 @@ def get_sarvam_language(language_code: Optional[str]) -> Language:
+        current_lang_code = self._settings.get("target_language_code", "en-IN")
+
+        if new_lang_code != current_lang_code:
+            logger.info(
+                f"[SARVAM] Script detected: {detected_script} - switching {current_lang_code} to {new_lang_code}"
+            )
+            self._settings["target_language_code"] = new_lang_code
+
+            try:
+                await self._send_config()
+                logger.info(f"[SARVAM] Language switched to {new_lang_code}")
+                return True
```

Ruff (0.14.10)

Consider moving this statement to an `else` block

(TRY300)